subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/ComponentState.java

```diff
@@ -165,11 +164,8 @@ public void restartIncomingEdges(ComponentState selected) {
         }
     }
 
-    @Override
-    public void selectedBy(ResolvableSelectorState selectorState) {
-        if (firstSelectedBy == null) {
-            firstSelectedBy = (SelectorState) selectorState;
-        }
```

Comment:
The behavior of this changed with the selector sorting. So a gaol of this PR is to get rid of this additional (weird) state and instead rely on the order of the selector list.